File: Examples/VisualBasic/AsposePDF/Pages/SplitToPages.vb
Imports System.IO
Imports System
Imports Aspose.Pdf

Namespace AsposePDF.Pages
    Public Class SplitToPages
        Public Shared Sub Run()
            ' ExStart:SplitToPages
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Pages()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("SplitToPages.pdf"))

            Dim pageCount As Integer = 1

            ' Loop through all the pages
            For Each pdfPage As Page In pdfDocument.Pages
                Dim newDocument As New Document()
                newDocument.Pages.Add(pdfPage)
                newDocument.Save((dataDir & Convert.ToString("page_")) + pageCount.ToString() + "_out_" + ".pdf")
                pageCount += 1
            Next
            ' ExEnd:SplitToPages       

        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePDF/Pages/SplitToPages.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;

namespace CSharp.AsposePdf.Pages
{
    public class SplitToPages
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Pages();
            
            //open document
            Document pdfDocument = new Document(dataDir + "SplitToPages.pdf");

            int pageCount = 1;

            //loop through all the pages
            foreach (Page pdfPage in pdfDocument.Pages)
            {
                Document newDocument = new Document();
                newDocument.Pages.Add(pdfPage);
                newDocument.Save(dataDir + "page_" + pageCount + ".pdf");
                pageCount++;
            }

            
        }
    }
}